marlowe-playground-client/src/Halogen/Monaco.purs

```diff
@@ -56,6 +56,8 @@ instance boundedEnumKeyBindings :: BoundedEnum KeyBindings where
 type State
   = { editor :: Maybe Editor
     , deactivateBindings :: Effect Unit
+    , getModelMarkers :: String -> Array IMarkerData
```

Comment:
I'm always wary of putting functions in state objects. Can you explain why it has to be a function here, please?

Reply:
In this case we want to be able to change that function over time. In our particular case we want to change the linting function to include the current simulation state but I've tried to avoid linking up Marlowe code with Monaco code so we don't want to just include the simulation state and pin `Halogen.Monaco` to be a Marlowe specific thing.

Reply:
ok, I have restructured things so that the linting is now done in Simulation.purs and `Halogen.Monaco` doesn't need to know anything special about getting model markers, it can simply set and get them.